data/src/main/java/com/example/data/repository/CalendarRepository.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.data.repository
+
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.example.data.database.dao.JobDao
+import com.example.data.database.entity.JobEntity
+import com.example.data.network.CalendarServer
+import com.example.data.network.setting.CalenderApi
+import com.example.domain.i_repository.ICalendarRepository
+import com.example.domain.model.calendar.Calendar
+import com.example.domain.model.calendar.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import retrofit2.Retrofit
+import javax.inject.Inject
+
+class CalendarRepository
+@Inject constructor(
+    @CalenderApi private val provideRetrofit: Retrofit,
+    private val providesJobDao: JobDao
+) : ICalendarRepository {
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override suspend fun findCalendar(): Flow<List<Calendar>> = flow {
+        val calendarListRes: List<Calendar> =
+            provideRetrofit.create(CalendarServer::class.java).fetchCalendar().body()!!
+                .map { it.toCalendar() }
+
+        val jobListRes: List<Job> = calendarListRes.flatMap { it.jobs }.toList()
+
+        val jobMap: Map<Int, JobEntity> = providesJobDao.getAllJobs().associateBy { it.id!! }
```

Comment:
Không khai báo kiểu khi không thực sự cần thiết

```suggestion
        val calendarListRes =
            provideRetrofit.create(CalendarServer::class.java).fetchCalendar().body()!!
                .map { it.toCalendar() }

        val jobListRes = calendarListRes.flatMap { it.jobs }.toList()

        val jobMap = providesJobDao.getAllJobs().associateBy { it.id!! }
```